File: backend/SRMS.API/Models/Route.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SRMS.API.Models;

[Table("routes")]
public class TrafficRoute
{
    [Key]
    [Column("route_id")]
    public Guid RouteId { get; set; }

    [Required]
    [MaxLength(200)]
    [Column("route_name")]
    public string RouteName { get; set; } = string.Empty;

    [Required]
    [MaxLength(200)]
    [Column("start_location")]
    public string StartLocation { get; set; } = string.Empty;

    [Required]
    [MaxLength(200)]
    [Column("destination")]
    public string Destination { get; set; } = string.Empty;

    [Required]
    [Column("distance_km")]
    public decimal DistanceKm { get; set; }

    [Required]
    [Column("estimated_time_minutes")]
    public int EstimatedTimeMinutes { get; set; }

    [Required]
    [MaxLength(50)]
    [Column("traffic_level")]
    public string TrafficLevel { get; set; } = string.Empty;

    [Column("created_at")]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    [Column("updated_at")]
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    // Navigation properties
    public virtual ICollection<EmergencySession> EmergencySessions { get; set; } = new List<EmergencySession>();
    public virtual ICollection<RouteJunction> RouteJunctions { get; set; } = new List<RouteJunction>();
}

File: mobile/lib/models/route.dart
import 'route_junction.dart';

class Route {
  final String routeId;
  final String routeName;
  final String startLocation;
  final String destination;
  final double distanceKm;
  final int estimatedTimeMinutes;
  final String trafficLevel;
  final List<RouteJunction> junctions;

  Route({
    required this.routeId,
    required this.routeName,
    required this.startLocation,
    required this.destination,
    required this.distanceKm,
    required this.estimatedTimeMinutes,
    required this.trafficLevel,
    required this.junctions,
  });

  factory Route.fromJson(Map<String, dynamic> json) {
    final junctionsList = json['junctions'] as List<dynamic>? ?? [];
    final junctions = junctionsList
        .map((junctionJson) => RouteJunction.fromJson(junctionJson as Map<String, dynamic>))
        .toList();

    return Route(
      routeId: json['routeId'] as String,
      routeName: json['routeName'] as String,
      startLocation: json['startLocation'] as String,
      destination: json['destination'] as String,
      distanceKm: (json['distanceKm'] as num).toDouble(),
      estimatedTimeMinutes: json['estimatedTimeMinutes'] as int,
      trafficLevel: json['trafficLevel'] as String,
      junctions: junctions,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'routeId': routeId,
      'routeName': routeName,
      'startLocation': startLocation,
      'destination': destination,
      'distanceKm': distanceKm,
      'estimatedTimeMinutes': estimatedTimeMinutes,
      'trafficLevel': trafficLevel,
      'junctions': junctions.map((j) => j.toJson()).toList(),
    };
  }
}
